### DISCUSSION TITLE
Doubt regarding Handers in fuzz tests

### DISCUSSION THREAD

**Discussion by heheboii11** [Cyfrin/foundry-full-course-cu#2912] [General]
Hey, In a doubt on how the sequence is done for Handlers in Fuzz tests. I am very confused, that in Patrics code he used the ```mint handler``` on the top of the code and then followed by ```deposit and redeem handlers```. But in invariants section, if we target ```Dscengine```, it will call functions very randomly with random inputs. Just wanna know how the sequencing works in the fuzz tests!!!

**Comment by EngrPips** [comment]
Hello @heheboii11, a Fuzz test is not meant to have a sequence of calling functions as the primary objective of a Fuzzer is to call random functions on your contract with random input, so if a Fuzzer then has a specific order for calling a function in your implementation contract, then it is no more a Fuzzer. I stand to be corrected.

**Reply by heheboii11** [reply]
But in patrics lesson, he mentioned that he used Handlers, such that it calls mintDSC or redeem collateral only after there is collateral deposited by the user. Can you explain this part ?

**Reply by EngrPips** [reply]
Did Patrick say he Instructed the Fuzzer to only call `mintDSC` and `redeemCollateral` after the Fuzzer had already called `depositCollateral`? Or he wrote some logic inside the `mintDSC` and `redeemCollateral` function of the Handler contract, which makes them return if no collateral has been deposited yet. 

You can share the link to the timestamp where Patrick said `he used Handlers, such that it calls mintDSC or redeem collateral only after there is collateral deposited by the user`
